src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java

```diff
@@ -76,16 +78,28 @@ public void onCreate(Bundle paramBundle) {
     bccSelfCheckbox.setOnPreferenceChangeListener((preference, newValue) -> {
       boolean enabled = (Boolean) newValue;
       dcContext.setConfigInt(CONFIG_BCC_SELF, enabled? 1 : 0);
+      DcHelper.getAccounts(getContext()).startIo();
       return true;
     });
 
     mvboxMoveCheckbox = (CheckBoxPreference) this.findPreference("pref_mvbox_move");
     mvboxMoveCheckbox.setOnPreferenceChangeListener((preference, newValue) -> {
       boolean enabled = (Boolean) newValue;
+      DcHelper.getAccounts(getContext()).stopIo();
       dcContext.setConfigInt(CONFIG_MVBOX_MOVE, enabled? 1 : 0);
+      DcHelper.getAccounts(getContext()).startIo();
       return true;
     });
 
+    onlyFetchMvboxCheckbox = this.findPreference("pref_only_fetch_mvbox");
+    onlyFetchMvboxCheckbox.setOnPreferenceChangeListener(((preference, newValue) -> {
+      boolean enabled = (Boolean) newValue;
+      DcHelper.getAccounts(getContext()).stopIo();
+      dcContext.setConfigInt(CONFIG_ONLY_FETCH_MVBOX, enabled? 1 : 0);
+      DcHelper.getAccounts(getContext()).startIo();
```

Comment:
we should probably add the line `changes require restarting IO by calling dc_stop_io() and then dc_start_io().` to deltachat.h to make the requirement clear.

Reply:
I'll do that